test/compilable/test19891.d

```diff
@@ -0,0 +1,7 @@
+void fun(R)(auto ref int a, auto ref R r = 1) {}
+
+void main()
+{
+    fun(10, 2); // Error: cannot modify constant `1`
+    fun(10); // Error: `auto` can only be used as part of `auto ref` for template function parameters
```

Comment:
This looks strange - it's the same error messages as in the bugzilla issue.

This test case should be in `fail_compilation`, and the generated error messages should be tested with `TEST_OUTPUT`, and a link to the bugzilla issue should be included.